frontend/src/components/organisms/ManageWebsite.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import Button from "../atoms/Button";
+import { api } from "@/utils/api";
+import Card from "../molecules/Card";
+
+const ManageWebsite = () => {
+  function convertToCSV(data: any[]) {
+    const csvArray = [];
+  
+    // Add header row
+    const headers = Object.keys(data[0]);
+    csvArray.push(headers.join(","));
+  
+    // Add data rows
+    data.forEach(item => {
+      const values = headers.map(header => item[header]);
+      csvArray.push(values.join(","));
+    });
+  
+    // Join rows with newline character
+    return csvArray.join("\n");
+  }
+
+  function createCSVFromArray(csvDataArray: any[]) {
+    // Combine all CSV data into one string
+    const combinedCSVData = csvDataArray.join('\n');
+  
+    // Create a Blob object from the combined CSV data
+    const blob = new Blob([combinedCSVData], { type: 'text/csv' });
+  
+    // Create a temporary anchor element to trigger the download
+    const anchor = document.createElement('a');
+    anchor.href = URL.createObjectURL(blob);
+    anchor.download = 'database.csv';
+    anchor.click();
+  
+    // Clean up
+    URL.revokeObjectURL(anchor.href);
+  }
+
+  const handleDownloadDatabase = async () => {
+
+    const eventsData = await api.get("/events/")
+    const events = eventsData.data.data.result
+
+    const enrollmentsData = await api.get("/events/enrollments") // new endpoint
+    const enrollments = enrollmentsData.data.data
+
+    const profilesData = await api.get("/users/profiles") // new endpoint
+    const profiles = profilesData.data.data
+
+    const permissionsData = await api.get("/users/permissions") // new endpoint but this table is empty
+    const permissions = permissionsData.data.data
+
+    const userData = await api.get("/users/")
+    const users = userData.data.data.result
+
+    const preferencesData = await api.get("/users/preferences") // new endpoint
+    const preferences = preferencesData.data.data
+
+    // Combine all data arrays
+    const allData = [events, enrollments, users, preferences];
+    const allCSVs = []
```

Comment:
Okay this is going to be a significant change to the overall design of this PR, but i think it is more ideal to keep the logic server side. What I mean is that instead of making different requests (multiple endpoints) you can simply use one endpoint (maybe /download or something) then in the server, you don't even need to make new endpoints - simply use Prisma and it's APIs to make the relevant queries. Then send the combined JSON back as a response

Please reach out if I am not clear enough

Reply:
I don't imagine there will be a size limit but we can see what the response looks like